cmd/lambda/accounts/list_test.go

```diff
@@ -1,173 +1,125 @@
 
-	"github.com/Optum/dce/pkg/api/response"
-	"github.com/Optum/dce/pkg/db"
-	"github.com/Optum/dce/pkg/db/mocks"
-	"github.com/aws/aws-lambda-go/events"
+	"github.com/Optum/dce/pkg/account"
+	"github.com/Optum/dce/pkg/account/accountiface/mocks"
+	"github.com/Optum/dce/pkg/config"
+	"github.com/gorilla/schema"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/mock"
 )
 
-func TestListController_Call(t *testing.T) {
+func TestGetAccounts(t *testing.T) {
 
-	t.Run("When the invoking Call and there are no errors", func(t *testing.T) {
-		expectedAccounts := createAccountsOutput()
-		mockAccountInput := createGetAccountsInput()
-		expectedAccountsResponse := &[]*response.AccountResponse{
-			{
-				ID:             "123456789",
-				AccountStatus:  "Ready",
-				LastModifiedOn: 1561149393,
-				CreatedOn:      1561149393,
-				AdminRoleArn:   "mock:role:arn",
+	type response struct {
+		StatusCode int
+		Body       string
+	}
+	tests := []struct {
+		name        string
+		expResp     response
+		expLink     string
+		query       *account.Account
+		retAccounts *account.Accounts
+		retErr      error
+		nextID      *string
+	}{
+		{
+			name:  "get all accounts",
+			query: &account.Account{},
+			expResp: response{
+				StatusCode: 200,
+				Body:       "[]\n",
 			},
-		}
-		mockDb := mocks.DBer{}
-		mockDb.On("GetAccounts", mockAccountInput).Return(*expectedAccounts, nil)
-		mockRequest := createGetSingleAccountRequest()
-
-		Dao = &mockDb
-
-		actualResponse, err := Handler(context.TODO(), mockRequest)
-		require.Nil(t, err)
-
-		parsedResponse := &[]*response.AccountResponse{}
-		err = json.Unmarshal([]byte(actualResponse.Body), parsedResponse)
-		require.Nil(t, err)
-
-		require.Equal(t, expectedAccountsResponse, parsedResponse, "it returns a list of accounts.")
-		require.Equal(t, actualResponse.StatusCode, 200, "it returns a 200.")
-	})
-
-	t.Run("When the query fails", func(t *testing.T) {
-		expectedError := errors.New("error")
-		mockAccountInput := createGetEmptyAccountsInput()
-		mockAccountOutput := createAccountsOutput()
-		mockDb := mocks.DBer{}
-		mockDb.On("GetAccounts", mockAccountInput).Return(*mockAccountOutput, expectedError)
-		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/accounts"}
-
-		Dao = &mockDb
-
-		actualResponse, err := Handler(context.TODO(), mockRequest)
-		require.Nil(t, err)
-
-		require.Equal(t, actualResponse.StatusCode, 500, "it returns a 500.")
-	})
-
-	t.Run("Building next URL", func(t *testing.T) {
-		body := strings.NewReader("")
-		request, err := http.NewRequest("GET", "http://example.com/api/accounts?limit=2", body)
-
-		assert.Nil(t, err)
-
-		nextParams := make(map[string]string)
-
-		nextParams["AccountId"] = "1"
-
-		nextURL := response.BuildNextURL(request, nextParams, url.URL{})
-
-		assert.Equal(t, url.Values{
-			"limit": {
-				"2",
+			retAccounts: &account.Accounts{},
+			retErr:      nil,
+		},
+		{
```

Comment:
Can you please add tests to get accounts with other query parameters accountStatus, adminRoleArn etc,

Reply:
These tests are written to handle code paths.  So my question is what are you looking to get from the tests?  Are we validating the Account struct can decode all the different query parameters?